tests/loan-token/TradingTestToken.test.js

```diff
@@ -792,5 +792,287 @@ contract("LoanTokenTrading", (accounts) => {
                 "entry price above the minimum"
             );
         });
+
+        it("Check marginTrade with minPositionSize > 0 (with defaultConversionPath set)", async () => {
+            // await set_demand_curve(loanToken);
+            await SUSD.transfer(loanToken.address, wei("1000000", "ether"));
+            await RBTC.transfer(accounts[2], oneEth);
+            await RBTC.approve(loanToken.address, oneEth, { from: accounts[2] });
+
+            await expectRevert(
+                loanToken.marginTrade(
+                    "0x0", // loanId  (0 for new loans)
+                    wei("2", "ether"), // leverageAmount
+                    0, // loanTokenSent (SUSD)
+                    10000, // collateral token sent
+                    RBTC.address, // collateralTokenAddress (RBTC)
+                    accounts[1], // trader,
+                    20000, // minEntryPrice
+                    "0x", // loanDataBytes (only required with ether)
+                    { from: accounts[2] }
+                ),
+                "principal too small"
+            );
+
+            await priceFeeds.setRates(
+                WRBTC.address,
+                SUSD.address,
+                new BN(10).pow(new BN(20)).toString()
+            );
+            await priceFeeds.setRates(
+                RBTC.address,
+                SUSD.address,
+                new BN(10).pow(new BN(20)).toString()
+            );
+
+            const defaultPathConversion = [SUSD.address, RBTC.address, RBTC.address];
```

Comment:
the conversion path should not have the same token as the source and dest

Reply:
Yeah, this is just a use for test purposes, in the mainnet, shouldn't be the case as in it will have 3 length of addresses at very minimum (especially for v1 pool)

Reply:
i mean i believe it should be restricted
else in case of an error it will work extremely inefficient and pretty hard to discover